UEFIExtract/ffsdumper.cpp

```diff
@@ -15,22 +15,52 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 #include <fstream>
 
+static void FixFileName(UString &name)
+{
+    // Replace some symbols with underscores for compatibility
+    const char table[] = {
+        '/', // Banned in *nix and Windows
+        '<', '>', ':', '\"', '\\', '|', '?', '*', // Banned in Windows
+    };
+    int nameLength = (int)name.length(); // Note: Qt uses int for whatever reason.
+    if (!nameLength) {
+        name += '_';
+    } else {
+        for (int i = 0; i < nameLength; i++) {
+            if (name[i] < 0x20 || name[i] > 0x7f) {
+               name[i] = '_';
+               continue;
+            }
+            for (size_t j = 0; j < sizeof(table); j++) {
+                if (name[i] == table[j]) {
+                    name[i] = '_';
+                    break;
+                }
+            }
+        }
+    }
+}
+
 USTATUS FfsDumper::dump(const UModelIndex & root, const UString & path, const DumpMode dumpMode, const UINT8 sectionType, const UString & guid)
 {
     dumped = false;
     counterHeader = counterBody = counterRaw = counterInfo = 0;
     fileList.clear();
 
-    if (changeDirectory(path))
+    if (changeDirectory(path)) {
+        fprintf(stdout, "Can't change directory %s\n", (const char*)path.toLocal8Bit());
```

Comment:
Case and presence/absence of ":" is inconsistent in this new error messages. 
Also, why not just printf(...) instead of fprintf(strdout, ...) ?

printf("Something is wrong: %u", error_code) should be the consistent format, IMO.

Reply:
fixed in new commit https://github.com/LongSoft/UEFITool/pull/241/commits/4c35a9d670ae0478424d5cc2e42ada424a0722e4